### ISSUE TITLE
Iterators do not work with unrolling

### ISSUE THREAD

**Issue by keram88** [keram88/smack#4]
For example:
for i in 0..2 { .... } with unroll=4 will unroll the loop 4 times.

**Comment by Guoanshisb** [comment]
Setting unroll to 4 is supposed to unroll the loop 4 times. The last iteration is blocked because the condition is evaluated to false. I guess what you mean is that the loop body is (symbolically) executed 4 times, right?

**Comment by keram88** [comment]
Yes. It was trying to look up array index 4. I've switched to while loops. Iterators will need to be dealt with.

**Comment by keram88** [comment]
Here is an example program
fn main() {
    for i in 0..4 {
        assert!(i < 4);
    }
}

This fails verification for unroll > 4. SMACK makes up the value i == 4294966627 to break the assertion.

**Comment by Guoanshisb** [comment]
@keram88 Could you attach the Boogie file to this issue?

**Comment by keram88** [comment]
bpl for simple iterator failing
[out.bpl.zip](https://github.com/keram88/smack/files/257945/out.bpl.zip)
